src/test/java/com/machowina/service/TicketServiceTest_calculatingFee.java

```diff
@@ -0,0 +1,135 @@
+		}
+
+	@Test
+	public void calculateFeeForGivenDuration_3_hours() {
+		//given
+		Duration duration = Duration.ofHours(3l);
+		//when
+		BigDecimal fee = ticketService.calculateFeeForGivenDuration(duration, rates);
+		//then
+		Assert.assertEquals(new BigDecimal("7.00"), fee);
+	}
+	
+	@Test
+	public void calculateFeeForGivenDuration_12_minutes() {
+		//given
+		Duration duration = Duration.ofMinutes(12l);
+		//when
+		BigDecimal fee = ticketService.calculateFeeForGivenDuration(duration, rates);
+		//then
+		Assert.assertEquals(new BigDecimal("0.20"), fee);
+	}
+	
+	@Test
+	public void calculateFeeForGivenDuration_5_hours_30_minutes() {
+		//given
+		Duration duration = Duration.ofMinutes(330l);
+		//when
+		BigDecimal fee = ticketService.calculateFeeForGivenDuration(duration, rates);
+		//then
+		Assert.assertEquals(new BigDecimal("47.00"), fee);
+	}
+	
+	@Test
+	public void calculateFeeForGivenDuration_VIP_20_minutes() {
+		//given
+		Duration duration = Duration.ofMinutes(20l);
+		//when
+		BigDecimal fee = ticketService.calculateFeeForGivenDuration(duration, vipRates);
+		//then
+		Assert.assertEquals(new BigDecimal("0.00"), fee);
+	}
+	
+	@Test
+	public void calculateFeeForGivenDuration_VIP_3_hours() {
+		//given
+		Duration duration = Duration.ofHours(3l);
+		//when
+		BigDecimal fee = ticketService.calculateFeeForGivenDuration(duration, vipRates);
+		//then
+		Assert.assertEquals(new BigDecimal("5.00"), fee);
+	}
+	
+	@Test
+	public void calculateTicketFee_3_hours() {
+		//given
+		User driver = new User("kowalski","pass","regular");
+		ParkingTicket ticket  = new ParkingTicket(LocalDateTime.now(), zone, null, driver);
+		ticket.setStopTime(LocalDateTime.now().plusHours(3l));
+		ticket.setStopped(true);
+		ticket.setId(1l);
+		Mockito.when(ticketRepository.findOne(1l)).thenReturn(ticket);
+		Mockito.when(parkingRatesService.findOneByDriverTypeAndParkingZone("regular", zone)).thenReturn(rates);
+		Mockito.when(ticketRepository.save(ticket)).thenReturn(ticket);
+		
+		//when
+		BigDecimal fee = ticketService.calculateTicketFee(1l);
+		//then
+		Assert.assertEquals(new BigDecimal("7.00"), fee);
+		Assert.assertEquals(ticket.getFee(), fee);
+	}
+	@Test(expected = TicketNotStoppedException.class)
+	public void calculateTicketFee_not_stopped() {
+		//given
+		User driver = new User("kowalski","pass","regular");
+		ParkingTicket ticket  = new ParkingTicket(LocalDateTime.now(), zone, null, driver);
+		ticket.setStopTime(LocalDateTime.now().plusHours(3l));
+		ticket.setId(1l);
+		Mockito.when(ticketRepository.findOne(1l)).thenReturn(ticket);
+		Mockito.when(parkingRatesService.findOneByDriverTypeAndParkingZone("regular", zone)).thenReturn(rates);
+		Mockito.when(ticketRepository.save(ticket)).thenReturn(ticket);
+		
+		//when
+		BigDecimal fee = ticketService.calculateTicketFee(1l);
+		//then catch exception
+		Assert.assertTrue(ticket.getFee()==null);
```

Comment:
nie ma innej metody w Assert która by to sprawdziła bez `==null`?

Reply:
assertNotNull:)